### PR TITLE
Guided Onboarding: Start the experiment

### PR DESCRIPTION
**Pull request by escapemanuele**
Related to 7021-gh-Automattic/dotcom-forge

## Proposed Changes

This PR:
- Removes the Cft structure
- Changes the flow name from 'guided' to 'onboarding'
- Checks for the experiment and uses the assignment
- Adds `trailmap_variant` to the event

## Why are these changes being made?

Starts the Guided Onboarding experiment! pbxNRc-3JR-p2

## Testing Instructions

- Go to the experiment page ( 21809-explat-experiment ) and assign yourself the appropriate group
- Test the flow
- Check that the experiment assignment is added to the page_view event

### CODE REVIEW

File: client/signup/controller.js [hunk 1]
```diff
@@ -233,31 +233,12 @@ export default {
 
 		// See: 1113-gh-Automattic/experimentation-platform for details.
 		if ( isOnboardingFlow || isOnboardingGuidedFlow( flowName ) ) {
```
Comment (heavyweight) [inline client/signup/controller.js:235]: These checks now are basically the same

Comment (escapemanuele) [reply client/signup/controller.js:235]: Let's keep them for now, as we may need to revert the change of guided to be 'onboarding'. 

https://github.com/Automattic/dotcom-forge/issues/7840

**Review by alshakero — APPROVED** [review]
## Testing report

### Control

1. No survey ✅ 
2. Free domain → Free plan card ✅
3. Paid domain → Free plan CTA ✅
4. Clicking free plan CTA shows a modal ✅
5. Continuing with Free plan and domain shows me the goals survey. ✅
6. Going to `https://SITE/wp-admin/options.php` shows no intent nor goals ✅

**Tracks**: `trailmap_variant: control` exists ✅

### Treatment

**Tracks**: `trailmap_variant: treatment_guided ` exists ✅

#### Blogger Segment

1. Got the survey ✅ 
2. Chose "Creating a site for myself, a business, or a friend", went to Q2 ✅ 
3. Selected "Publish a blog", chose a free domain, plan grid has Free Starter Explorer Creator ✅
3. Selected "Publish a blog", chose a paid domain, plan grid has Starter Explorer Creator with a CTA for free plan ✅
4. Selected "Starter plan", paid, skipped Email upsell, landed in `site-setup-wg/options` ✅
5. Site intent is `build` ✅ 
6. Site goals are `[ "write" ]`  ✅

#### Merchant Segment

1. Got the survey ✅ 
2. Chose "Creating a site for myself, a business, or a friend", went to Q2 ✅ 
3. Selected "Sell something", chose a free domain, plan grid has Creator Entrepreneur Enterprise ✅
4. Selected "Creator plan", paid, landed in `site-setup-wg/options` ✅
5. Site intent is `build` ✅ 
6. Site goals are `[  "sell" ]`  ✅

#### Consumer Segment

1. Got the survey ✅ 
2. Chose "Creating a site for myself, a business, or a friend", went to Q2 ✅ 
3. Selected "Promote by business", chose a free domain, plan grid has Starter Explorer Creator ✅
4. Selected "Creator plan", paid, landed in `site-setup-wg/options` ✅
5. Site intent is `build` ✅ 
6. Site goals are `[  "promote" ]`  ✅

#### Developer / Agency

1. Got the survey ✅ 
2. Chose "Creating a site client", **skipped** Q2 ✅ 
3. Plan grid only has Creator Entrepreneur Enterprise. ✅
4. Selected "Creator plan", paid, landed in `site-setup-wg/goals` ⚠️  (is this intentional?).
5. Site intent is `host-site` ✅ 
6. Site goals are `[ ]`  ✅
8. How do I verify that the user was marked as a dev?

#### Unknown

1. Got the survey ✅ 
2. Skipped Q1, landed in domains (not in Q2, is that intentional?)
3. Picked free domain, got all the plans ✅
4. Picked paid domain, got all plans, with de-emphasized free plan ✅

#### Newsletter
1. Picked newsletter.
2. Got redirected to `setup/newsletter/newsletterSetup?ref=guided-onboarding` ✅
6. Created a newsletter and landed in Launchpad.

### Treatment - Survey only

**Tracks**: `trailmap_variant: treatment_survey_only ` exists ✅

1. Got the survey ✅ 
7. Chose "Creating a site for myself, a business, or a friend", went to Q2 ✅ 
9. Selected "Sell something", chose a free domain, plan grid has all the plans ✅
10. Selected "Free", landed in `site-setup-wg/options` ✅
11. Site intent is `build` ✅ 
12. Site goals are `[  "sell" ]`  ✅

## Bugs found - Not necessarily in the scope of this PR
1. The back button is duplicated in Q2 of the survey (there are two stacked buttons).
2. The back button is duplicated in the domain step too.

**Comment by escapemanuele** [comment]
Thank you @alshakero!

> Skipped Q1, landed in domains (not in Q2, is that intentional?)

Yes! As from the PT:
```
Users who selected “Creating a site for myself, a business, or a friend” will be taken to Question 2.
```

**Review by aaronyan — APPROVED** [review]
LGTM! Thank you for getting this ready for next week @escapemanuele and thank you for the thorough testing @alshakero!

**Pull request merged**